### PR TITLE
Implement starter code for Honor System.

### PR DESCRIPTION
**Pull request by calebkrauter**
Intention is for the honor system to modify stats of the player's character based on milestones called honorEventMarkers. BAD CHAD events can also happen depending upon the marker of bad honor.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -26,6 +26,8 @@
     <script type="text/javascript" src="./js/classes/inventory.js"></script>
     <script type="text/javascript" src="./js/classes/timer.js"></script>
     <script type="text/javascript" src="./js/classes/vector.js"></script>
+    <script type="text/javascript" src="js\scripts\honorSystem\honorSystem.js"></script>
```
Comment (b1gd3vd0g) [inline index.html:29]: please try to match the style of your filepaths to all the other ones. Also, let's avoid capital letters in the filepaths (recommended for JS).

File: js/scripts/honorSystem/honorSystem.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ * Initializes the honor at the start of a game to default honor 
+ * unless honor values already exists in currentHonor.
+ * This class is intended to modify some of CHAD's stats based on his honor and the honor points gained
+ * or lost during a particular event or action. 
+ * The constructor takes in a positive or negative value for delatahonorPts from at least 0 to 5 points.
+ * @author Caleb Krauter
+ */
+
+class honorSystem {
+    constructor(deltaHonorPts, isNewGame) {
```
Comment (b1gd3vd0g) [inline js/scripts/honorSystem/honorSystem.js:11]: Constructor needs proper JSDoc commenting. It is unclear what type deltaHonorPts and isNewGame should be, nor do I understand what they should represent.

File: js/scripts/honorSystem/honorSystem.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            this.currentHonor = this.INITIIAL_HONOR;
+        } else {
+            // Not sure if this works.
+            this.currentHonor = this.currentHonor;
+        }
+        manageHonor(this.currentHonor, deltaHonorPts);
+    }
+
+    /** CHAD's initial honor at the start of the game. */
+    get INITIIAL_HONOR() {
```
Comment (b1gd3vd0g) [inline js/scripts/honorSystem/honorSystem.js:36]: This should be static.

**Review by b1gd3vd0g — COMMENTED** [review]
I'm not approving it - I think it is preemptive. I think we'll be more ready to start implementing this system when we have more of a game to play.

**Review by tclaar — COMMENTED** [review]
The honor system probably should not have its own class given the limited functionality it currently contains. In addition, I like the design of the system this class outlines and definitely think it would be nice to add it at some point, but it seems too early to implement since we don't yet have the features it is intended to work with (e.g. player character health).

**Comment by calebkrauter** [comment]
> I'm not approving it - I think it is preemptive. I think we'll be more ready to start implementing this system when we have more of a game to play.

Sounds good! You are definitely correct and I appreciate your response! Looking forward to implementing this later.

**Comment by calebkrauter** [comment]
> The honor system probably should not have its own class given the limited functionality it currently contains. In addition, I like the design of the system this class outlines and definitely think it would be nice to add it at some point, but it seems too early to implement since we don't yet have the features it is intended to work with (e.g. player character health).

Sounds good! Thanks for the detailed response. You're right, we can put this into CHAD's class. However I am a little bit concerned that CHAD's class (Currently PapaChad.js) will become very messy and would like us to consider creating a directory for different classes that make up CHAD and if the Honor System becomes large enough perhaps it will be its own class. For now I agree and when I implement it later I will plan to implement it directly into CHAD's class.

**Pull request closed without merging**